### PR TITLE
#166187823 User should be able to see Articles Paginated.

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do?

Add Pagination support for project

#### Description of Task to be completed?

- Add provision for pagination to view all articles
- Add pagination for search results

#### How should this be manually tested?

* -	Git fetch this repo locally in the terminal with `git fetch origin ft-article-pagination-166187823
* -	Run application: `npm start`
* -	Navigate to the: `/articles` route
The review app is [here.](https://authors-frontend-staging-pr-23.herokuapp.com/articles)

#### Any background context you want to provide?

- Have added pagination support for the project so that 6 stories are shown per page to prevent excessive scrolling..

#### What are the relevant pivotal tracker stories?

#[#166187823](https://www.pivotaltracker.com/story/show/#166187823)

#### Screenshots:

Screenshot 2019-07-24 at 19 04 02

Screenshot 2019-07-24 at 19 04 10

Screenshot 2019-07-24 at 19 04 22

### CODE REVIEW

**Review by HabibSentongo — APPROVED** [review]
Wow, this works perfectly well, good work team. I give this a green light.

**Review by alexxsanya — CHANGES_REQUESTED** [review]
Screenshot 2019-07-25 at 17 50 00

@neelxie well done on this though I notice the `1/1 page `label looks mis-asligned. I suggest you style it with 
```CSS
line-height: 3.2;
font-family: Lato;
font-weight: 200;
```
and we should good to go.

**Review by Emmanuelaaron — APPROVED** [review]
Everything seems working well on my side.
great work @neelxie

**Comment by neelxie** [comment]
> Screenshot 2019-07-25 at 17 50 00
> 
> @neelxie well done on this though I notice the `1/1 page `label looks mis-asligned. I suggest you style it with
> 
> ```css
> line-height: 3.2;
> font-family: Lato;
> font-weight: 200;
> ```
> 
> and we should good to go.

Thank you @alexxsanya I have worked on this feedback, thank you.

**Review by alexxsanya — APPROVED** [review]

**Review by jennizalwango — APPROVED** [review]
Good work at derrick l have tested out your PR and  everything works well on my side

File: src/Articles/components/AllArticles.jsx [hunk 1]
```diff
@@ -5,17 +5,36 @@ import Article from "../../commons/components/Article";
...
   componentDidMount() {
     const { getArticles } = this.props;
     getArticles();
   }
+
+  changePage = (apiCallUrl, changeIt ) => {
+    const { getArticles } = this.props;
+    const { pageNumber } = this.state;
+    this.setState({
+      pageNumber: pageNumber + changeIt
```
Comment (AGMETEOR) [inline src/Articles/components/AllArticles.jsx]: Better name for `changeIt` arg

File: src/Articles/tests/components/paginationComponent.test.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    nxtButton.simulate('click', event);
+    expect(props.paginate).toBeCalled();
+    prevButton.simulate('click', event);
+
+
+    props.previous = 'previous';
+    wrapper = mount(<Pagination {...props} />);
+    wrapper.find(Button).at(0).simulate('click', event);
+    expect(props.paginate).toBeCalled();
```
Comment (AGMETEOR) [inline src/Articles/tests/components/paginationComponent.test.jsx:47]: Test which page we're on based on a previous page number...

File: src/auth/signup/components/SignUp.jsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { connect } from 'react-redux';
 import signUpUser from '../redux/actions/signUpAction';
 import validation from '../Validation';
 import '../CSS/signup.scss';
+import FacebookButton from '../../socialAuth/facebook/components/Facebook';
```
Comment (AGMETEOR) [inline src/auth/signup/components/SignUp.jsx]: Are these changes relevant to the story of this PR?

**Review by AGMETEOR — CHANGES_REQUESTED** [review]

File: src/auth/signup/components/SignUp.jsx (hunk 3 above)
Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: Yes they are because if you check develop they are not there.

Comment (AGMETEOR) [reply src/auth/signup/components/SignUp.jsx]: How is social auth related to pagination?

Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: The social-auth button on the develop signup were not functional, you can check it out, and this is the fix for them.

Comment (AGMETEOR) [reply src/auth/signup/components/SignUp.jsx]: If they're not working, does that mean you make changes under a story that is in no way related to social auth?

Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: Am sorry about that, I should have sought advice on handling that.

Comment (AGMETEOR) [reply src/auth/signup/components/SignUp.jsx]: What's the better way to handle that?

Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: I guess it is to seek guidance on how to handle it.

Comment (AGMETEOR) [reply src/auth/signup/components/SignUp.jsx]: Please ask some of the team members, I need to be confident that you really understand how to handle it...

Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: Yes @AGMETEOR I can and let me work on that.

Comment (neelxie) [reply src/auth/signup/components/SignUp.jsx]: I have removed it all and fixed this problem.

**Review by AGMETEOR — APPROVED** [review]
